packages/firestore/src/model/mutation.ts

```diff
@@ -689,14 +675,6 @@ function localTransformResults(
       previousValue = maybeDoc.field(fieldTransform.field);
     }
 
-    if (previousValue === null && baseDoc instanceof Document) {
```

Comment:
Why is this logic no longer needed? This seems pretty critical. Or is it handled elsewhere now?

Reply:
My understanding is that this if-statement checks for the edge case where a PatchMutation clears the values before the TransformMutation can be applied. With the addition of updateTransforms, field transformation are encoded on the Patch or Set Mutation, rather than in a separate TransformMutation. This means that we won't bump into this edge case anymore since the update and fieldTransform are on the same mutation.

I've also removed the now obsolete server timestamp test that tested this edge case.